cmd/aigw/run_test.go

```diff
@@ -31,218 +32,116 @@ import (
 
 	"github.com/envoyproxy/ai-gateway/cmd/extproc/mainlib"
 	"github.com/envoyproxy/ai-gateway/internal/filterapi"
-	internaltesting "github.com/envoyproxy/ai-gateway/internal/testing"
 )
 
-// setupDefaultAIGatewayResourcesWithAvailableCredentials sets up the default AI Gateway resources with available
-// credentials and returns the path to the resources file and the credentials context.
-func setupDefaultAIGatewayResourcesWithAvailableCredentials(t *testing.T) (string, internaltesting.CredentialsContext) {
-	credCtx := internaltesting.RequireNewCredentialsContext()
-	// Set up the credential substitution.
-	t.Setenv("OPENAI_API_KEY", credCtx.OpenAIAPIKey)
-	aiGatewayResourcesPath := filepath.Join(t.TempDir(), "ai-gateway-resources.yaml")
-	awsCredTmpFile := filepath.Join(t.TempDir(), "aws-credentials")
-	err := os.WriteFile(awsCredTmpFile, []byte(credCtx.AWSFileLiteral), 0o600)
-	require.NoError(t, err)
-	aiGatewayResources := strings.ReplaceAll(aiGatewayDefaultResources, "~/.aws/credentials", awsCredTmpFile)
-	err = os.WriteFile(aiGatewayResourcesPath, []byte(aiGatewayResources), 0o600)
-	require.NoError(t, err)
-	return aiGatewayResourcesPath, credCtx
-}
-
 func TestRun(t *testing.T) {
-	resourcePath, cc := setupDefaultAIGatewayResourcesWithAvailableCredentials(t)
+	ollamaModel := getOllamaChatModel(t)
+	if ollamaModel == "" || !checkIfOllamaReady(t, ollamaModel) {
+		t.Skipf("Ollama not ready or model %q missing. Run 'ollama pull %s' if needed.", ollamaModel, ollamaModel)
+	}
+
+	t.Setenv("OPENAI_BASE_URL", "http://localhost:11434/v1")
+	t.Setenv("OPENAI_API_KEY", "unused")
+
 	ctx, cancel := context.WithCancel(t.Context())
 	defer cancel()
 	done := make(chan struct{})
 	go func() {
 		opts := runOpts{extProcLauncher: mainlib.Main}
-		require.NoError(t, run(ctx, cmdRun{Debug: true, Path: resourcePath}, opts, os.Stdout, os.Stderr))
+		require.NoError(t, run(ctx, cmdRun{Debug: true}, opts, os.Stdout, os.Stderr))
 		close(done)
 	}()
-	defer func() {
-		// Make sure the external processor is stopped regardless of the test result.
-		cancel()
-		<-done
-	}()
-
-	// This is the health checking to see the envoy admin is working as expected.
-	require.Eventually(t, func() bool {
-		req, err := http.NewRequestWithContext(ctx, http.MethodGet, "http://localhost:9901/ready",
```

Comment:
our implicit config doesn't currently start an admin server, so that's why this is no longer here